src/main/java/com/redhat/rhjmc/containerjfr/net/web/handlers/RecordingUploadPostHandler.java

```diff
@@ -0,0 +1,174 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * #L%
+ */
+package com.redhat.rhjmc.containerjfr.net.web.handlers;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import com.redhat.rhjmc.containerjfr.MainModule;
+import com.redhat.rhjmc.containerjfr.core.sys.Environment;
+import com.redhat.rhjmc.containerjfr.core.sys.FileSystem;
+import com.redhat.rhjmc.containerjfr.net.AuthManager;
+import com.redhat.rhjmc.containerjfr.net.web.HttpMimeType;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+import io.vertx.ext.web.multipart.MultipartForm;
+
+class RecordingUploadPostHandler extends AbstractAuthenticatedRequestHandler {
+
+    private final Environment env;
+    private final WebClient webClient;
+    private final FileSystem fs;
+    private final Path savedRecordingsPath;
+
+    @Inject
+    RecordingUploadPostHandler(
+            AuthManager auth,
+            Environment env,
+            WebClient webClient,
+            FileSystem fs,
+            @Named(MainModule.RECORDINGS_PATH) Path savedRecordingsPath) {
+        super(auth);
+        this.env = env;
+        this.webClient = webClient;
+        this.fs = fs;
+        this.savedRecordingsPath = savedRecordingsPath;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.POST;
+    }
+
+    @Override
+    public String path() {
+        return "/api/v1/recordings/:recordingName/upload";
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    void handleAuthenticated(RoutingContext ctx) throws Exception {
+        String recordingName = ctx.pathParam("recordingName");
+        try {
+            URL uploadUrl = new URL(env.getEnv("GRAFANA_DATASOURCE_URL"));
+            // TODO validate URL before POST attempt
+            ResponseMessage response = doPost(recordingName, uploadUrl);
+            ctx.response().setStatusCode(response.statusCode);
+            ctx.response().setStatusMessage(response.statusMessage);
+            ctx.response().end(response.body);
+        } catch (MalformedURLException e) {
+            throw new HttpStatusException(501, e);
+        }
+    }
+
+    // FindBugs thinks the recordingPath or its properties is null somehow
```

Comment:
Would the check for this be running `mvn spotbugs:check` ?

Reply:
Er, with the suppression removed :P

Reply:
Yea that should do it

Reply:
Dang, guess our projects really do differ in some critical way. I've removed the suppression and see green for my runs of `spotbugs:check`

Reply:
s/project/systems I suppose

Reply:
I don't get the Spotbugs warning anymore either - maybe I had come back around at some point and refactored this in a way that avoids the previous problem. I'll remove the annotation.

Reply:
I can see why it would have this warning since there are calls that would NPE if their input was null and the values submitted aren't checked before being used as input. However it looks like the values are currently never not null and I guess it would have trouble analyzing that? Anyways; weird that I don't see the warning on my runs...